### ISSUE TITLE
[defect]: radiusd v. 3.2.5 crashes several times a day with SIGSEGV, Segmentation fault.

### ISSUE THREAD

**Issue by sistemits** [FreeRADIUS/freeradius-server#5383]
### What type of defect/bug is this?

Crash or memory corruption (segv, abort, etc...)

### How can the issue be reproduced?

Can't reproduce, it happens 3-5 times a day.

According to the core dump, the problem is that request->packet is null.

file: "src/main/stats.c", line number: 99, symbol: "request_stats_final"
91      void request_stats_final(REQUEST *request)
92      {
93              rad_listen_t *listener;
94              RADCLIENT *client;
95
96              if ((request->options & RAD_REQUEST_OPTION_STATS) != 0) return;
97
98              /* don't count statistic requests */
99              if (request->packet->code == PW_CODE_STATUS_SERVER) {
100                     return;
101             }
102
103             listener = request->listener;

(gdb) print request
$3 = (REQUEST *) 0x7fc3f032cc10
(gdb) print request->packet
$4 = (RADIUS_PACKET *) 0x0

### Log output from the FreeRADIUS daemon

```shell
radiusd.service: Main process exited, code=dumped, status=11/SEGV
systemd[1]: radiusd.service: Failed with result 'core-dump'.
```

### Backtrace from LLDB or GDB

```shell
(gdb) bt
#0  0x000000000042ea74 in request_stats_final (
    request=request@entry=0x7fc3f032cc10) at src/main/stats.c:99
#1  0x0000000000436601 in request_stats_final (request=0x7fc3f032cc10)
    at src/main/stats.c:91
#2  request_done (request=request@entry=0x7fc3f032cc10,
    original=original@entry=2) at src/main/process.c:921
#3  0x0000000000438abc in request_ping (request=0x7fc3f032cc10,
    action=<optimized out>) at src/main/process.c:3973
#4  0x0000000000434241 in request_proxy_reply (packet=0x7fc3f81dc9f0)
    at src/main/process.c:3013
#5  0x000000000041fe8d in proxy_socket_recv (listener=0x52c8e70)
    at src/main/listen.c:2639
#6  0x00007fc409586714 in fr_event_loop (el=0x26516a0) at src/lib/event.c:713
#7  0x00000000004135cc in main (argc=<optimized out>, argv=<optimized out>)
    at src/main/radiusd.c:641
(gdb) bt full
#0  0x000000000042ea74 in request_stats_final (
    request=request@entry=0x7fc3f032cc10) at src/main/stats.c:99
        listener = <optimized out>
        client = <optimized out>
        auth_stats = <optimized out>
        coa_stats = <optimized out>
#1  0x0000000000436601 in request_stats_final (request=0x7fc3f032cc10)
    at src/main/stats.c:91
        listener = <optimized out>
        client = <optimized out>
        auth_stats = <optimized out>
        coa_stats = <optimized out>
        dsc_stats = <optimized out>
        done = <optimized out>
        proxy_stats = <optimized out>
#2  request_done (request=request@entry=0x7fc3f032cc10,
    original=original@entry=2) at src/main/process.c:921
        now = {tv_sec = 140479530183792, tv_usec = -815144620468448512}
        when = {tv_sec = 40179360, tv_usec = 1}
        action = <optimized out>
#3  0x0000000000438abc in request_ping (request=0x7fc3f032cc10,
    action=<optimized out>) at src/main/process.c:3973
        home = <optimized out>
        buffer = "10.252.29.20\000\000\000\000\002\000\000\000\n\374\035\024\260\213\n\354\303\177\000\000\000\000\000\000 \000\000\000\000\000\000\000R\267\025\af\000\000\000\000\000\000\000\360\060I\t\304\177\000\000\260҇\002\000\000\000\000\f\311X\t\304\177\000\000\245\313\300\312\376P\232-\237\256\205\"\037v\351Ǡ\026e\002\000\000\000\000\000ShXI\006\260\364\301\256X\t\304\177\000\000x\314\062\360\303\177\000"
        __FUNCTION__ = "request_ping"
#4  0x0000000000434241 in request_proxy_reply (packet=0x7fc3f81dc9f0)
    at src/main/process.c:3013
        proxy_p = 0x7fc3f032cc78
        request = 0x7fc3f032cc10
        now = {tv_sec = 1721765547, tv_usec = 494133}
        buffer = "\270\026e\002\000\000\000\000\360J{\002\377\177\000\000\230\030e\002\000\000\000\000`4I\t\304\177\000\000\220\311\035\370\303\177\000\000\000ShXI\006\260\364\253\016\240f\000\000\000\000\000ShXI\006\260\364\000\000\000\000\000\000\000\000J", '\000' <repeats 15 times>, "\001\000\000\000\000\000\000\000\270\026e\002\000\000\000\000\360J{\002\377\177\000\000\230\030e\002\000\000\000\000\340\367W\t\304\177\000"
#5  0x000000000041fe8d in proxy_socket_recv (listener=0x52c8e70)
    at src/main/listen.c:2639
        packet = 0x7fc3f81dc9f0
        sock = <optimized out>
        buffer = "\337\310X\t\304\177\000\000\024&I\t\304\177\000\000p\254~\360\303\177\000\000\030\000\000\000\000\000\000\000\270\030e\002\000\000\000\000\270Ӈ\002\000\000\000\000\240\026e\002\000\000\000\000@B\320\b\304\177\000\000\200J{\002\377\177", '\000' <repeats 18 times>, "\252\262\262\023\000\000\000\000\240--Type <RET> for more, q to quit, c to continue without paging--
\026e\002\000\000\000\000\000ShXI\006\260\364\270\026e\002\000\000\000\000\240\026e\002\000\000\000"
#6  0x00007fc409586714 in fr_event_loop (el=0x26516a0) at src/lib/event.c:713
        ef = 0x2651898
        i = 6
        rcode = <optimized out>
        when = {tv_sec = 1721765547, tv_usec = 824571}
        wake = <optimized out>
        read_fds = {fds_bits = {0, 1024, 0 <repeats 14 times>}}
        write_fds = {fds_bits = {0 <repeats 16 times>}}
#7  0x00000000004135cc in main (argc=<optimized out>, argv=<optimized out>)
    at src/main/radiusd.c:641
        rcode = 0
        status = <optimized out>
        argval = <optimized out>
        spawn_flag = true
        display_version = false
        flag = 0
        from_child = {-1, -1}
        p = <optimized out>
        state = 0x481840 <global_state.lto_priv>
        autofree = 0x2205620
```

**Comment by manu0401** [comment]
This has been fixed in https://github.com/FreeRADIUS/freeradius-server/commit/3a9449539e4c5a74c85685cad6abe6edf412f701

**Comment by alandekok** [comment]
We will release 3.2.6 shortly.

**Comment by miker985** [comment]
I think I ran into the same issue with 3.0.27

```
==1622== Invalid read of size 4
==1622==    at 0x434AF6: request_stats_final (stats.c:99)
==1622==    by 0x43E21C: request_done (process.c:921)
==1622==    by 0x44437E: request_ping (process.c:3788)
==1622==    by 0x442642: request_proxy_reply (process.c:2903)
==1622==    by 0x41AD3D: proxy_socket_recv (listen.c:2275)
==1622==    by 0x4476BC: event_socket_handler (process.c:5159)
==1622==    by 0x50A7889: fr_event_loop (event.c:715)
==1622==    by 0x449261: radius_event_process (process.c:6294)
==1622==    by 0x43333D: main (radiusd.c:641)
==1622==  Address 0x44 is not stack'd, malloc'd or (recently) free'd
==1622==
==1622==
==1622== Process terminating with default action of signal 11 (SIGSEGV)
==1622==  Access not within mapped region at address 0x44
==1622==    at 0x434AF6: request_stats_final (stats.c:99)
==1622==    by 0x43E21C: request_done (process.c:921)
==1622==    by 0x44437E: request_ping (process.c:3788)
==1622==    by 0x442642: request_proxy_reply (process.c:2903)
==1622==    by 0x41AD3D: proxy_socket_recv (listen.c:2275)
==1622==    by 0x4476BC: event_socket_handler (process.c:5159)
==1622==    by 0x50A7889: fr_event_loop (event.c:715)
==1622==    by 0x449261: radius_event_process (process.c:6294)
==1622==    by 0x43333D: main (radiusd.c:641)
==1622==  If you believe this happened as a result of a stack
==1622==  overflow in your program's main thread (unlikely but
==1622==  possible), you can try to increase the size of the
==1622==  main thread stack using the --main-stacksize= flag.
==1622==  The main thread stack size used in this run was 8388608.
==1622==
==1622== HEAP SUMMARY:
==1622==     in use at exit: 80,306,063 bytes in 719,957 blocks
==1622==   total heap usage: 3,544,811 allocs, 2,824,854 frees, 787,665,453 bytes allocated
==1622==
==1622== LEAK SUMMARY:
==1622==    definitely lost: 5,448 bytes in 55 blocks
==1622==    indirectly lost: 5,882 bytes in 55 blocks
==1622==      possibly lost: 7,475,643 bytes in 6,875 blocks
==1622==    still reachable: 72,819,090 bytes in 712,972 blocks
==1622==         suppressed: 0 bytes in 0 blocks
==1622== Rerun with --leak-check=full to see details of leaked memory
==1622==
==1622== For lists of detected and suppressed errors, rerun with: -s
==1622== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

I've applied the changes @manu0401 linked and things have run fine for several hours (previous record with 3.0.27 was < 30 minutes)

Thank you!

**Comment by alandekok** [comment]
@miker985 I've pushed a fix for v3.0.x, thanks.
